calyx/src/passes/canonical.rs

```diff
@@ -59,7 +59,15 @@ impl Visitor for Canonicalize {
     ) -> VisResult {
         comp.for_each_assignment(|assign| {
             if let Guard::Port(p) = &(*assign.guard) {
+                /* Trying to cover r1.done? 1'd1.
+                if p.borrow().name == "done"
+                    && assign.src.borrow().clone().is_constant(1, 1)
+                {
+                    assign.src = p.clone();
+                    assign.guard = Guard::True.into()
+                }
                 // 1'd1 ? r1.done
+                else*/
```

Comment:
Why is this code commented out?

Reply:
I was thinking of handling the `a? 1'd1 -> a ` case, but we discussed that you can't do that. I should've deleted this before I made the PR, sorry.